src/implementation/c/node/table-lookup.ts

```diff
@@ -183,6 +185,91 @@ export class TableLookup extends Node<frontend.node.TableLookup> {
     return true;
   }
 
+  private buildNeon(out: string[]): boolean {
+    const ctx = this.compilation;
+
+    const edge = this.ref.edges[0];
+    assert(edge !== undefined);
+
+    const ranges = this.buildRanges(edge);
+
+    if (ranges.length === 0) {
+      return false;
+    }
+
+    // Way too many calls would be required
+    if (ranges.length > MAX_NEON_RANGES) {
+      return false;
+    }
+
+    out.push('#ifdef __ARM_NEON__');
+    out.push(`while (${ctx.endPosArg()} - ${ctx.posArg()} >= 16) {`);
+    out.push('  uint8x16_t input;');
+    out.push('  uint8x16_t single;');
+    out.push('  uint8x16_t mask;');
+    out.push('  uint8x8_t narrow;');
+    out.push('  uint64_t match_mask;');
+    out.push('  int match_len;');
+    out.push('');
+    out.push('  /* Load input */');
+    out.push(`  input = vld1q_u8(${ctx.posArg()});`);
+
+    out.push('  /* Find first character that does not match `ranges` */');
+    function v128(value: number): string {
+      return `vdupq_n_u8(${ctx.toChar(value)})`;
+    }
+
+    for (let off = 0; off < ranges.length; off += 2) {
+      const start = ranges[off];
+      const end = ranges[off + 1];
+      assert(start !== undefined);
+      assert(end !== undefined);
+
+      // Same character, equality is sufficient (and faster)
+      if (start === end) {
+        out.push(`  single = vceqq_u8(input, ${v128(start)});`);
+      } else {
+        out.push(`  single = vandq_u16(`);
+        out.push(`    vcgeq_u8(input, ${v128(start)}),`);
+        out.push(`    vcleq_u8(input, ${v128(end)})`);
+        out.push('  );');
+      }
+
+      if (off === 0) {
+        out.push('  mask = single;');
+      } else {
+        out.push('  mask = vorrq_u16(mask, single);');
```

Comment:
`vorrq_u16` expects both arguments to be `uint16x8_t` but both are `uint8x16_t`.